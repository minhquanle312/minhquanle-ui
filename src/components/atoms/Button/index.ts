import { Button as AntdButton } from './Button';

export const Button = AntdButton;
