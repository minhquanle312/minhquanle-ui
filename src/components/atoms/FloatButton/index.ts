import { FloatButton as AntdFloatButton } from './FloatButton';

export const FloatButton = AntdFloatButton;
