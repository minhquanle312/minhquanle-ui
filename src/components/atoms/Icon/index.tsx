// Libraries
import styled, { css } from 'styled-components'

// Components
import AntsIcon from '@antscorp/icons'

// Constants
import { THEME } from 'minhquanle-ui/es/constants'

interface IconProps {
  size?: number
  width?: number
  height?: number
  color?: string
  style?: React.CSSProperties
}

const DraftAntsIcon = AntsIcon as any

export const Icon = styled(DraftAntsIcon)<IconProps>`
  ${(p) => (p.style ? css`style` : null)}
  ${(p) =>
    p.size
      ? css`
          width: ${p.width || p.size}px !important;
          height: ${p.height || p.size}px !important;
          font-size: ${p.size}px !important;
        `
      : css`
          font-size: 20px;
          line-height: 1;
        `}

  ${(p) =>
    p.color
      ? css`
          color: ${p.color} !important;
        `
      : null}

  ${(p) =>
    p.disabled
      ? css`
          cursor: not-allowed !important;
          color: ${THEME.token?.colorTextDisabled} !important;
        `
      : null}
`
