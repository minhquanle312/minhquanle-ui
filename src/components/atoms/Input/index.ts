export { Input, TextArea, InputProps } from './Input';
