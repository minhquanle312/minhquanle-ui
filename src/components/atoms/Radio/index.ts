// Libraries
import styled from 'styled-components'

// Components
import { Radio as AntdRadio } from 'antd'

// Constants
import { THEME } from 'minhquanle-ui/lib/constants'

export const Radio = styled(AntdRadio)`
  font-family: 'Roboto';
  color: ${THEME.token?.colorTextBase};
  font-size: ${THEME.token?.fontSize}px;

  .antsomi-radio {
    .antsomi-radio-inner {
      border-width: 2px;
      border-color: ${THEME.token?.colorPrimary};

      &::after {
        background-color: ${THEME.token?.colorPrimary};
      }
    }
  }
`

export const RadioButton = styled(AntdRadio.Button)``
export const RadioGroupSub = styled(AntdRadio.Group)``
