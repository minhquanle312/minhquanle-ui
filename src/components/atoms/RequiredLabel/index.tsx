// Libraries
import React, { memo, ReactNode } from 'react';

// Atoms
import { Text, TextProps } from '../Text';

// Styled
import { StyledRequired } from './styled';

interface RequiredLabelProps extends TextProps {
  children?: ReactNode;
}

export const RequiredLabel: React.FC<RequiredLabelProps> = memo(props => {
  // Props
  const { children, ...restProps } = props;

  return (
    <Text {...restProps}>
      <StyledRequired>*</StyledRequired>
      {children}
    </Text>
  );
});
