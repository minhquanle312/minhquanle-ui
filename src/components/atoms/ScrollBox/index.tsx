/* eslint-disable @typescript-eslint/no-unused-expressions */
import React, { useState, useEffect, useRef } from 'react';
import styled from 'styled-components';

// Constants
import { THEME } from 'src/constants';

interface ScrollBoxProps extends React.HTMLAttributes<HTMLDivElement> {
  height?: number;
  maxHeight?: number;
  loadMore?: Function;
  isPadding?: boolean;
}

export const ScrollBox: React.FC<ScrollBoxProps> = props => {
  const { children, height, maxHeight, isPadding, loadMore, style, ...restOf } = props;
  const wrapperRef = useRef<HTMLDivElement>(null);
  const [isScroll, setIsScroll] = useState(false);

  useEffect(() => {
    if (wrapperRef.current && wrapperRef.current.scrollHeight > wrapperRef.current.clientHeight) {
      !isScroll && setIsScroll(true);
    } else {
      isScroll && setIsScroll(false);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [wrapperRef.current]);

  const handleScroll = e => {
    const bottom = e.target.scrollTop + e.target.clientHeight >= e.target.scrollHeight;

    if (bottom) {
      if (typeof loadMore === 'function') {
        loadMore();
      }
    }
  };

  return (
    // eslint-disable-next-line @typescript-eslint/no-use-before-define
    <Wrapper
      ref={wrapperRef}
      {...{
        ...restOf,
        style: {
          ...style,
          ...(height && { height }),
          ...(maxHeight && { maxHeight }),
          paddingRight: isScroll && isPadding ? 8 : 0,
        },
      }}
      onScroll={handleScroll}
    >
      {children}
    </Wrapper>
  );
};

const Wrapper = styled.div`
  height: 100%;
  width: 100%;
  background-color: ${THEME.token?.bw0};
  overflow: auto;
`;

ScrollBox.defaultProps = {
  isPadding: false,
};
