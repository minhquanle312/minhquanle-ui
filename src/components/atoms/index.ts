export { Button } from './Button'
export { Divider } from './Divider'
export { DividerPure } from './DividerPure'
export { Input, TextArea } from './Input'
export { InputNumber } from './InputNumber'
export { Radio, RadioGroupSub, RadioButton } from './Radio'
export { Space } from './Space'
export { Tag } from './Tag'
export { Tooltip } from './Tooltip'
export { Typography } from './Typography'
export { Checkbox } from './Checkbox'
export { message } from './Message'
export { Segmented } from './Segmented'
export { Timeline } from './Timeline'
export { Row, Col } from './Grid'
export { FloatButton } from './FloatButton'
export { Skeleton } from './Skeleton'
export { notification } from './Notification'
export { Progress } from './Progress'
export { Pagination } from './Pagination'
export { Steps } from './Steps'
export { Empty } from './Empty'
export { Icon } from './Icon'
export { Text } from './Text'
export { Popover } from './Popover'
export { RequiredLabel } from './RequiredLabel'
export { Switch } from './Switch'
export { Slider } from './Slider'
export { Spin } from './Spin'
export { Alert } from './Alert'
export { ScrollBox } from './ScrollBox'

// Export Types
export type { SliderProps } from './Slider'
