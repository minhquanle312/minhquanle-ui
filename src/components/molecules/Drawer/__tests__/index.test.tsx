// Libraries
import React, { useState } from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

// Components
import { Drawer } from 'antd';
import { Button } from '../../../atoms';

const TestDrawer = () => {
  const [open, setOpen] = useState(false);

  const showDrawer = () => {
    setOpen(true);
  };

  const onClose = () => {
    setOpen(false);
  };

  return (
    <>
      <Button type="primary" onClick={showDrawer}>
        Open
      </Button>
      <Drawer title="Basic Drawer" onClose={onClose} open={open}>
        <p>Some</p>
        <p>Some</p>
        <p>Some</p>
      </Drawer>
    </>
  );
};

describe('<Drawer />', () => {
  it('renders Default Drawer Component', async () => {
    render(<TestDrawer />);

    const openButton = screen.getByText('Open');
    expect(openButton).toBeInTheDocument();

    fireEvent.click(openButton);

    const drawerTitle = await screen.getByText('Basic Drawer');
    expect(drawerTitle).toBeInTheDocument();

    const drawerContent = await screen.findAllByText('Some');
    expect(drawerContent.length).toBe(3);
  });
  it('closes the drawer when clicking button', async () => {
    render(<TestDrawer />);

    const openButton = screen.getByText('Open');
    fireEvent.click(openButton);

    const drawerTitle = screen.queryByText('Basic Drawer');
    expect(drawerTitle).not.toBeEmptyDOMElement();
  });
  it('should match snapshot', () => {
    const loadingIndicator = render(<TestDrawer />);
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
