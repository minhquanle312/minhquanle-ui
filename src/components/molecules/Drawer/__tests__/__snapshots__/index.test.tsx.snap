// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> should match snapshot 1`] = `
<button
  class="ant-btn css-dev-only-do-not-override-htwhyh ant-btn-primary"
  type="button"
>
  <span>
    Open
  </span>
</button>
`;
